test/e2e/daemon_set.go

```diff
@@ -597,13 +598,24 @@ func checkDaemonPodsImage(c clientset.Interface, ns string, selector map[string]
 		}
 		pods := podList.Items
 
+		allImagesUpdated := true
 		for _, pod := range pods {
 			podImage := pod.Spec.Containers[0].Image
-			if podImage != image || !v1.IsPodReady(&pod) {
-				framework.Logf("Wrong image for pod: %s. Expected: %s, got: %s. Pod Ready: %t", pod.Name, image, podImage, v1.IsPodReady(&pod))
-				return false, nil
+			if podImage != image {
+				allImagesUpdated = false
+				framework.Logf("Wrong image for pod: %s. Expected: %s, got: %s.", pod.Name, image, podImage)
+			}
+			if !v1.IsPodReady(&pod) {
+				framework.Logf("Pod %s is not ready", pod.Name)
+				unavailablePods++
 			}
 		}
+		if unavailablePods > maxUnavailable {
+			return false, fmt.Errorf("Error, number of unavailable pods: %d is greater than maxUnavailable: %d", unavailablePods, maxUnavailable)
```

Comment:
Remove "Error, " from the message - also error messages in fmt.Errorf / errors.New as a convention start with lowercased letter